pkgs/applications/version-management/forgejo/default.nix

```diff
@@ -1,16 +1,32 @@
-{ fetchurl, gitea, lib }:
+{ lib, fetchurl, gitea
+, pamSupport ? true
+, sqliteSupport ? true
+}:
 
 gitea.overrideAttrs (old: rec {
   pname = "forgejo";
-  version = "1.18.0-rc1-1";
+  version = "1.18.0-1";
 
   src = fetchurl {
     name = "${pname}-src-${version}.tar.gz";
     # see https://codeberg.org/forgejo/forgejo/releases
-    url = "https://codeberg.org/attachments/976c426a-3e04-49ff-9762-47fab50624a3";
-    hash = "sha256-kreBMHlMVB1UeG67zMbszGrgjaROateCRswH7GrKnEw=";
+    url = "https://codeberg.org/attachments/c829784c-3b85-4996-9dc6-09e12e40a93a";
+    hash = "sha256-42bR18T5ATVyhPejeHpOTkeP6VwYu5HFfei8oMjRJy8=";
   };
 
+  preBuild =
+    let
+      tags = lib.optional pamSupport "pam"
+        ++ lib.optional sqliteSupport "sqlite sqlite_unlock_notify";
+      tagsString = lib.concatStringsSep " " tags;
+    in
+    ''
+      export buildFlagsArray=(
+        -tags="${tagsString}"
+        -ldflags='-X "main.Version=${version}" -X "main.Tags=${tagsString}"'
```

Comment:
`ldflags` are canonically provided via attr: (meaning you would need to move your let to the top level)
```nix
ldflags = [
  "-X "main.Version=${version}"
  "-X main.Tags=${tagsString}"
];
```